src/chat/__tests__/chatSelectors-test.js

```diff
@@ -153,6 +158,110 @@ describe('getRecentConversations', () => {
   });
 });
 
+describe('getCurrentTypingUsers', () => {
+  test('return undefined when current narrow is not private or group', () => {
+    const state = {
+      chat: {
+        narrow: [],
+      },
+    };
+    const typingUsers = getCurrentTypingUsers(state);
+
+    expect(typingUsers).toEqual(undefined);
+  });
+
+  test('when in private narrow and the same user is typing return details', () => {
+    const expectedUser = {
+      id: 1,
+      email: 'john@example.com',
+      avatarUrl: 'http://example.com/avatar.png',
+      fullName: 'John Doe',
+    };
+    const state = {
+      accounts: [{ email: 'me@example.com' }],
+      chat: {
+        narrow: privateNarrow('john@example.com'),
+      },
+      typing: {
+        'john@example.com': [1],
+      },
+      users: [expectedUser],
+    };
+
+    const typingUsers = getCurrentTypingUsers(state);
+
+    expect(typingUsers).toEqual([expectedUser]);
+  });
+
+  test('when two people are typing, return details for all of them', () => {
+    const user1 = {
+      id: 1,
+      email: 'john@example.com',
+      avatarUrl: 'http://example.com/avatar1.png',
+      fullName: 'John Doe',
+    };
+    const user2 = {
+      id: 2,
+      email: 'mark@example.com',
+      avatarUrl: 'http://example.com/avatar2.png',
+      fullName: 'Mark Dark',
+    };
+
+    const state = {
+      accounts: [{ email: 'me@example.com' }],
+      chat: {
+        narrow: privateNarrow('john@example.com'),
+      },
+      typing: {
+        'john@example.com': [1, 2],
```

Comment:
didn't understand this test case; how can `mark@example.com` user id be added to typing state with `john@example.com` as recipient? (Assuming self email is `me@example.com`)